src/store.cc

```diff
@@ -772,6 +777,16 @@ StoreEntry::write (StoreIOBuffer writeBuffer)
     invokeHandlers();
 }
 
+void
+StoreEntry::writeData(StoreIOBuffer writeBuffer)
+{
+    assert(mem_obj);
+    debugs(20, 6, "offset " << writeBuffer.offset << " len " << writeBuffer.length);
+    auto &dataHdr = mem_obj->data_hdr;
+    assert (dataHdr.endOffset() || writeBuffer.offset == 0);
+    assert(dataHdr.write(writeBuffer));
```

Comment:
If this method survives, please respect NDEBUG builds, even though Squid has a few other bugs like this already.